### PR TITLE
Adjust SettingsTabs according to group role

### PR DESCRIPTION
**Pull request by Julian702**
This is how the GroupSettingsTab looks like for a non-admin user.
image

### CODE REVIEW

**Review by memeToasty — APPROVED** [review]

**Review by darmiel — CHANGES_REQUESTED** [review]
Just some smoll cosmetic things

**Review by 4KevR — APPROVED** [review]
I have nothing to add to the addressed points of @darmiel

File: frontend/components/navigation/Tabs/DeckSettingsTab.tsx [hunk 1]
```diff
@@ -36,14 +38,20 @@ export const DeckSettingsTab = ({
 	const [deckState, setDeck] = useState(deck);
 	const [isConfirmDeletion, setConfirmDelete] = useState(false);
 
+	const admin = group.groupRole
+		? groupRole[group.groupRole] >= groupRole.ADMIN
+		: false;
+
```
[the remark this answers is not in the document]
Comment (Julian702) [reply frontend/components/navigation/Tabs/DeckSettingsTab.tsx:44]: Needs an exclamation mark cause groupRole could be undefined.

File: frontend/components/form/FormButton.tsx [hunk 2]
```diff
@@ -29,7 +29,9 @@ interface FormButtonProps {
 
 const getStyle = {
 	primary:
-		"border-2 border-kiokuDarkBlue bg-kiokuDarkBlue font-semibold text-eggshell",
+		"bg-kiokuDarkBlue border-kiokuDarkBlue text-eggshell hover:scale-105 hover:cursor-pointer",
+	disabled:
+		"bg-gray-500 border-gray-500 text-eggshell hover:cursor-not-allowed",
```
[the remark this answers is not in the document]
Comment (Julian702) [reply frontend/components/form/FormButton.tsx]: changed in ade302ef239c3f26966610301da7b48f1a48232d

File: frontend/components/form/FormButton.tsx [hunk 3]
```diff
@@ -50,7 +52,7 @@ export const FormButton = ({
 	return (
 		<input
 			type="submit"
-			className={`flex justify-center rounded-md  text-center  shadow-sm outline-none transition hover:scale-105 hover:cursor-pointer ${getStyle[style]} ${className} ${getSize[size]}`}
+			className={`flex justify-center rounded-md border-2 text-center font-semibold shadow-sm outline-none transition ${getStyle[style]} ${className} ${getSize[size]}`}
```
[the remark this answers is not in the document]
Comment (Julian702) [reply frontend/components/form/FormButton.tsx]: fixed in ade302ef239c3f26966610301da7b48f1a48232d

File: frontend/components/form/FormButton.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Julian702) [reply frontend/components/form/FormButton.tsx]: fixed in ade302ef239c3f26966610301da7b48f1a48232d

File: frontend/components/input/Button.tsx [hunk 4]
```diff
@@ -28,10 +28,15 @@ interface ButtonProps {
...
-	warning: "bg-kiokuYellow border-kiokuYellow text-white",
+	primary:
+		"bg-kiokuDarkBlue border-kiokuDarkBlue text-eggshell shadow-sm hover:scale-105 hover:cursor-pointer",
+	secondary:
+		"bg-transparent border-transparent text-kiokuDarkBlue hover:scale-105 hover:cursor-pointer",
+	error: "bg-kiokuRed border-kiokuRed text-white hover:scale-105 hover:cursor-pointer",
+	warning:
+		"bg-kiokuYellow border-kiokuYellow text-white hover:scale-105 hover:cursor-pointer",
+	disabled:
+		"bg-gray-500 border-gray-500 text-eggshell hover:cursor-not-allowed",
```
[the remark this answers is not in the document]
Comment (Julian702) [reply frontend/components/input/Button.tsx]: fixed in ade302ef239c3f26966610301da7b48f1a48232d

File: frontend/components/input/Button.tsx [hunk 5]
```diff
@@ -28,10 +28,15 @@ interface ButtonProps {
...
 
 const getStyle = {
-	primary: "bg-kiokuDarkBlue border-kiokuDarkBlue text-eggshell shadow-sm",
-	secondary: "bg-transparent border-transparent text-kiokuDarkBlue",
-	error: "bg-kiokuRed border-kiokuRed text-white",
-	warning: "bg-kiokuYellow border-kiokuYellow text-white",
+	primary:
+		"bg-kiokuDarkBlue border-kiokuDarkBlue text-eggshell shadow-sm hover:scale-105 hover:cursor-pointer",
+	secondary:
+		"bg-transparent border-transparent text-kiokuDarkBlue hover:scale-105 hover:cursor-pointer",
```
[the remark this answers is not in the document]
Comment (Julian702) [reply frontend/components/input/Button.tsx]: added in ade302ef239c3f26966610301da7b48f1a48232d

File: frontend/components/input/DangerAction.tsx [hunk 6]
```diff
@@ -60,7 +65,7 @@ export const DangerAction = ({
 				button={
 					<Button
 						id={`${id}ButtonId`}
-						style="error"
+						style={disabled ? "disabled" : "error"}
 						size="sm"
 						className="h-fit w-full sm:w-1/3 md:w-1/4 lg:w-1/5 xl:w-1/6"
 						onClick={onClick}
```
[the remark this answers is not in the document]
Comment (Julian702) [reply frontend/components/input/DangerAction.tsx]: fixed in ade302ef239c3f26966610301da7b48f1a48232d

File: frontend/components/navigation/Tabs/DeckSettingsTab.tsx [hunk 7]
```diff
@@ -61,24 +69,23 @@ export const DeckSettingsTab = ({
...
+								description="Transfer this deck to another group where you have	the ability to create decks."
+								button="Transfer Deck"
+							></DangerAction>
+							<hr className="border-kiokuLightBlue" /> */}
 				<DangerAction
 					id="deleteDeckDangerAction"
 					header="Delete this deck"
-					description="Once you delete a deck, there is no going back. Please be
-					certain."
+					description="Once you delete a deck, there is no going back. Please be certain."
```
[the remark this answers is not in the document]
Comment (Julian702) [reply frontend/components/navigation/Tabs/DeckSettingsTab.tsx]: paraphrased in ade302ef239c3f26966610301da7b48f1a48232d

File: frontend/components/navigation/Tabs/GroupSettingsTab.tsx [hunk 8]
```diff
@@ -88,9 +109,9 @@ export const GroupSettingsTab = ({
 				<DangerAction
 					id="deleteGroupDangerAction"
 					header="Delete this group"
-					description="Once you delete a group, there is no going back. Please be
-					certain."
+					description="Once you delete a group, there is no going back. Please be certain."
```
[the remark this answers is not in the document]
Comment (Julian702) [reply frontend/components/navigation/Tabs/GroupSettingsTab.tsx]: paraphrased in ade302ef239c3f26966610301da7b48f1a48232d

File: frontend/components/navigation/Tabs/GroupSettingsTab.tsx [hunk 9]
```diff
@@ -57,11 +62,14 @@ export const GroupSettingsTab = ({
...
 					button="Save"
+					disabled={!admin}
 					onChange={(event: ChangeEvent<HTMLInputElement>) => {
 						setGroupDescription(event.target.value);
 					}}
 					onClick={() => {
-						modifyGroup({ groupDescription: groupDescription });
+						modifyGroup({
+							groupDescription: groupDescription,
+						});
```
[the remark this answers is not in the document]
Comment (Julian702) [reply frontend/components/navigation/Tabs/GroupSettingsTab.tsx]: fixed in ade302ef239c3f26966610301da7b48f1a48232d

**Review by darmiel — APPROVED** [review]

**Pull request merged**